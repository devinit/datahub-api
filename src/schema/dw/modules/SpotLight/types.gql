

type OverViewTabRegional {
  # WHAT PERCENTAGE OF PEOPLE IN WAKISO LIVE BELOW THE NATIONAL POVERTY LINE? 
  # can be no data or '12%'
  poorestPeople: String
  # WHAT RESOURCES ARE AVAILABLE TO LOCAL GOVERNMENTS IN WAKISO? eg 3.6m or 2.7bn
  # this is a total of local, donor and central government resources
  regionalResources: String
  # IndicatorDataColored is defined in country profile types
  # has local government, donor and central government
  regionalResourcesBreakdown: [IndicatorDataColored]
  # HOW MUCH DOES THE LOCAL GOVERNMENT SPEND PER PERSON?
  localGovernmentSpendPerPerson: String
}

type PovertyTabRegional {
  poorestPeople: String
  #WHAT IS THE AVERAGE LIFE EXPECTANCY? 
  lifeExpectancy: String
  #WHAT IS THE STANDARD OF LIVING SCORE? 
  stdOfLiving: String
}



type PopulationTabRegional {
  #The total population of a given district and the population density in per sq km
  totalPopulation: String
  populationDensity: String
   # Urban vs Rural population level
  populationDistribution: [PopulationDistribution]
  averageDependencyRatio: String
  allAverageDependencyRatio: String
}

type EducationTabRegional {
  #WHAT IS THE PUPILâ€“TEACHER RATIO IN PRIMARY EDUCATION?...in government schools  and...in all schools 
  pupilTeacherRatioGovtSchl: String
  pupilTeacherRatioOtherSchl: String
  #WHAT PERCENTAGE OF STUDENTS PASS THE PRIMARY LEAVING EXAM? 
  studentsPass: String
  #HOW MUCH PRIMARY EDUCATION FUNDING IS THERE? 
  primaryEducationfunding: String
}

type HealthTabRegional {
  #WHAT IS THE DISTRICT LEAGUE HEALTH PERFORMANCE SCORE?
  districtPerformance: String
  #WHAT PERCENTAGE OF TUBERCULOSIS CASES HAVE BEEN SUCCESSFULLY TREATED? 
  treatmeantOfTb: String
  #HOW MUCH LOCAL GOVERNMENT HEALTHCARE FUNDING IS THERE? 
  healthCareFunding: String
}


type LocalGovernmentFinance {
  # come from finance file
  revenue: [Domestic]
  expenditure: [Domestic]
}

type Query {  
  # id is district id
  getOverViewTab(id: String!):OverViewTabRegional
  getPovertyTab(id: String!): PovertyTabRegional
  getPopulationTab(id: String!):PopulationTabRegional
  getEducationTab(id: String!): EducationTabRegional
  getHealthTab(id: String!): HealthTabRegional 
  getLocalGovernmentFinance(id: String!): LocalGovernmentFinance
}

