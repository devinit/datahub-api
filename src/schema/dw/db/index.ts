import {IMain, IDatabase, IOptions} from 'pg-promise';
import {dwConfig} from '../config';
import Maps from '../modules/Maps';
import UnbundlingAid from '../modules/UnbundlingAid';
import CountryProfile from '../modules/CountryProfile';
import diagnostics from './diagnostics';
import * as pgPromise from 'pg-promise';
import * as LRU from 'lru-cache';
import {queue} from '../../../lib/cache';

export interface IExtensions {
    maps: Maps;
    unbundlingAid: UnbundlingAid;
    countryProfile: CountryProfile;
    manyCacheable: (query: string, values: any) => Promise<any>;
}

const lruOpts: LRU.Options<any> = {
    max: 500,
    maxAge: 1000 * 60 * 60 * 60
};

export const dbCache: LRU.Cache<any> = LRU(lruOpts);

// pg-promise initialization options:
const options: IOptions<IExtensions> = {
    // Extending the database protocol with our custom modules
    // API: http://vitaly-t.github.io/pg-promise/global.html#event:extend
    extend: (obj: IExtensions & IDatabase<IExtensions>) => {
        obj.manyCacheable = (query: string, values?: any) => {
            const getQuery = values ? pgPromise.as.format(query, values) : query;
            if (dbCache.has(getQuery)) {
                // add to queue so that we always have freshest data
                // makes same query in 15 minutes so as to update cache
                queue(getQuery, 'dw', dbCache, obj.many);
                return Promise.resolve(dbCache.get(getQuery));
            }
            return obj.any(getQuery);
        };
        obj.maps = new Maps(obj);
        obj.countryProfile = new CountryProfile(obj);
        obj.unbundlingAid = new UnbundlingAid(obj);
    },
    // caching
    receive: (data, _result, event) => {
        // cache recieved
        if (!dbCache.has(event.query)) dbCache.set(event.query, data);
    }
};

const pgp: IMain = pgPromise(options);

// Create the database instance with extensions:

const db = pgp(dwConfig) as IDatabase<IExtensions> & IExtensions;

// Load and initialize optional diagnostics:

diagnostics.init(options);

process.on('exit', (code) => {
  // kill db
  pgp.end();
  console.log(`About to exit with code: ${code}, closed DB connection`);
});

process.on('SIGINT', () => {
    // kill db
    pgp.end();
    console.log('Ctrl-C... forced termination closed DB connection');
});

// If you ever need access to the library's root (pgp object), you can do it via db.$config.pgp
// See: http://vitaly-t.github.io/pg-promise/Database.html#.$config
export default db;
