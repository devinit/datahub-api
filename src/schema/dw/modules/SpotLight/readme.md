Spotlight module focusing on sub regional API data, the maps API are handeled in the Maps module.