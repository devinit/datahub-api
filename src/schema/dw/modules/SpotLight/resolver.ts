import {
    IContext
} from '../../../../schema';

export default {
    Query: {
        async educationTabRegional(_root, args, ctx: IContext) {
            return ctx.dw.spotLight.getEducationTabRegional(args);
        },
        async healthTabRegional(_root, args, ctx: IContext) {
            return ctx.dw.spotLight.getHealthTabRegional(args);
        },
        async localGovernmentFinance(_root, args, ctx: IContext) {
            return ctx.dw.spotLight.getLocalGovernmentFinance(args);
        },
        async overviewTabRegional(_root, args, ctx: IContext) {
            return ctx.dw.spotLight.getOverViewTabRegional(args);
        },
        async populationTabRegional(_root, args, ctx: IContext) {
            return ctx.dw.spotLight.getPopulationTabRegional(args);
        },
        async povertyTabRegional(_root, args, ctx: IContext) {
            return ctx.dw.spotLight.getPovertyTabRegional(args);
        }
    }
};
