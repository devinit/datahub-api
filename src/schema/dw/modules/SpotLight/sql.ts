export default {
    // tslint:disable-next-line:max-line-length
    poorestPeople: 'SELECT * FROM ${schema^}.${country^}_poverty_headcount WHERE district_id = ${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    localGovernmentSpendPerPerson: 'SELECT * FROM ${schema^}.${country^}_gov_spend_pp WHERE district_id = ${id} AND year = ${start_year}',
    stdOfLiving: 'SELECT * FROM ${schema^}.${country^}_deprivation_living WHERE district_id = ${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    lifeExpectancy: 'SELECT * FROM ${schema^}.${country^}_life_expectancy WHERE district_id = ${id} AND year = ${start_year}',
    totalPopulation: 'SELECT * FROM ${schema^}.${country^}_total_pop WHERE district_id = ${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    populationDensity: 'SELECT * FROM ${schema^}.${country^}_pop_dens WHERE district_id = ${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    populationDistribution: 'SELECT * FROM ${schema^}.${country^}_population_rural_urban WHERE district_id = ${id} AND year >= ${start_year} AND year <= ${end_year}',
    averageDependencyRatio: 'SELECT * FROM ${schema^}.${country^}_dependency_ratio WHERE district_id = ${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    allAverageDependencyRatio: 'SELECT AVG(value) as value FROM ${schema^}.${country^}_dependency_ratio WHERE year = ${start_year}',
    pupilTeacherRatioGovtSchl: 'SELECT value FROM ${schema^}.${country^}_primary_stu_teach_ratio_gov WHERE district_id =${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    pupilTeacherRatioOtherSchl: 'SELECT value FROM ${schema^}.${country^}_primary_stu_teach_ratio WHERE district_id =${id} AND year = ${start_year}',
    studentsPassRate: 'SELECT value FROM ${schema^}.${country^}_leaving_exam_perf_rate WHERE district_id =${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    studentsPassDistrictRank: 'SELECT * FROM ${schema^}.${country^}_leaving_exam_perf_rate WHERE year = ${start_year} Order By value DESC',
    // tslint:disable-next-line:max-line-length
    primaryEducationfunding: 'SELECT value, value_ncu FROM ${schema^}.${country^}_primary_educ_funding WHERE district_id =${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    districtHealthPerformance: 'SELECT value FROM ${schema^}.${country^}_overall_health WHERE district_id =${id} AND year = ${end_year}',
    districtHealthPerformanceRank: 'SELECT * FROM ${schema^}.${country^}_overall_health WHERE year = ${end_year}  Order By value DESC',
    // tslint:disable-next-line:max-line-length
    treatmeantOfTb: 'SELECT value FROM ${schema^}.${country^}_tb_success WHERE district_id =${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    healthCareFunding: 'SELECT value, value_ncu FROM ${schema^}.${country^}_health_funding WHERE district_id =${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    lGFResources: 'SELECT value, value_ncu FROM ${schema^}.${country^}_igf_resources WHERE district_id =${id} AND year = ${start_year}',
    crResources: 'SELECT value, value_ncu FROM ${schema^}.${country^}_central_resources WHERE district_id =${id} AND year = ${start_year}',
    // tslint:disable-next-line:max-line-length
    dResources: 'SELECT value, value_ncu FROM ${schema^}.${country^}_donor_resources WHERE district_id = ${id} AND year = ${start_year}',
    localGovernmentFinance: 'SELECT * FROM ${schema^}.${country^}_finance WHERE district_id = ${id} AND l1 = ${l1} AND year >= ${start_year} AND value is NOT NULL'
};
