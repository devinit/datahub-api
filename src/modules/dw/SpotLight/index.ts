import { IDB } from '../../../api/db';
import { ISpotlightArgs, getOverViewTabRegional } from './lib/utils';
import Uganda from './lib/uganda';
import Kenya from './lib/kenya';

export default class SpotLight {
    public uganda: Uganda;
    public kenya: Kenya;
    public getOverViewTabRegional: (args: ISpotlightArgs) => Promise<DH.IOverviewTabRegional>;
    constructor(db: IDB) {
        this.uganda = new Uganda(db);
        this.kenya = new Kenya(db);
        this.getOverViewTabRegional = getOverViewTabRegional(db);
    }
}
