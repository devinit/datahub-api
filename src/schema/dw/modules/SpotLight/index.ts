import {IDatabase} from 'pg-promise';
import {IExtensions} from '../../db';
import {formatNumbers} from '../../../../utils';
import sql from './sql';
import {getConceptAsync, IConcept} from '../../../cms/modules/concept';
import * as R from 'ramda';
import {isError} from '../../../../lib/isType';
import {getIndicatorData, RECIPIENT, DONOR, IGetIndicatorArgs, isDonor, IProcessedSimple,
        IRAWPopulationAgeBand, normalizeKeyName, IRAW, IRAWQuintile, getTableNameFromSql,
        indicatorDataProcessingSimple, getTotal, IRAWPopulationGroup, } from '../utils';

interface ISpotlightArgs {
    id: string;
    country: string;
}

interface IRegionalResources {
    regionalResources: string;
    regionalResourcesBreakdown: DH.IIndicatorDataColored[];
}

export default class Profile {
    private db: IDatabase<IExtensions> & IExtensions;
    private defaultArgs;

    constructor(db: any) {
        this.db = db;
        this.defaultArgs = {db: this.db, conceptType: 'spotlight'};
    }
    // id eg uganda or kenya
    public async getOverViewTabRegional(opts: ISpotlightArgs): Promise<DH.IOverViewTabRegional> {
        const regionalResources = await this.getRegionalResources(opts);
        const [poorestPeople, localGovernmentSpendPerPerson] = await
            this.getIndicatorsGeneric(opts, [sql.poorestPeople, sql.localGovernmentSpendPerPerson]);
        return {
            ...regionalResources,
            poorestPeople,
            localGovernmentSpendPerPerson
        };
    }
    public async getPopulationTabRegional(opts: ISpotlightArgs): Promise<DH.IPopulationTabRegional> {
        const [totalPopulation, populationDensity, averageDependencyRatio, allAverageDependencyRatio]
             = await this.getIndicatorsGeneric(opts,
                [sql.totalPopulation, sql.populationDensity,
                sql.averageDependencyRatio, sql.allAverageDependencyRatio]);
        const populationDistribution = await this.getPopulationDistribution(opts);
        return {
            totalPopulation,
            populationDensity,
            populationDistribution,
            averageDependencyRatio,
            allAverageDependencyRatio
        };
    }
    public async getPovertyTabRegional(opts: ISpotlightArgs): Promise<DH.IPovertyTabRegional> {
        const [poorestPeople, lifeExpectancy, stdOfLiving] =
            await this.getIndicatorsGeneric(opts, [sql.poorestPeople, sql.lifeExpectancy, sql.stdOfLiving]);
        return {
            poorestPeople,
            lifeExpectancy,
            stdOfLiving
        };
    }
    public async getEducationTabRegional(opts: ISpotlightArgs): Promise<DH.IEducationTabRegional> {
        const [pupilTeacherRatioGovtSchl, pupilTeacherRatioOtherSchl, studentsPass, primaryEducationfunding] =
            await this.getIndicatorsGeneric(opts,
                [sql.pupilTeacherRatioGovtSchl, sql.pupilTeacherRatioOtherSchl,
                sql.studentsPass, sql.primaryEducationfunding]);
        return {
            pupilTeacherRatioGovtSchl,
            pupilTeacherRatioOtherSchl,
            studentsPass,
            primaryEducationfunding
        };
    }
    public async getHealthTabRegional(opts): Promise<DH.IHealthTabRegional> {
        const [districtPerformance, treatmeantOfTb, healthCareFunding] =
            await this.getIndicatorsGeneric(opts,
                [sql.districtHealthPerformance, sql.treatmeantOfTb, sql.healthCareFunding]);
        return {
            districtPerformance,
            treatmeantOfTb,
            healthCareFunding
        };
    }

    public async getLocalGovernmentFinance(id): Promise<DH.ILocalGovernmentFinance> {
        const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultArgs,
            query: sql.localGovernmentFinance,
            id
        };
        const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);

    }
    private async getRegionalResources(opts): Promise<IRegionalResources> {
        const indicatorArgs = [sql.lGFResources, sql.crResources, sql.dResources]
            .map(query => ({query, ...this.defaultArgs, ...opts}));
        const resourcesRaw: IRAW[][] = await Promise.all(indicatorArgs.map(args => getIndicatorData<IRAW>(args)));
        const resourcesSum = resourcesRaw.reduce((sum, data) => Number(data[0].value) + sum, 0);
        const resourceWithConceptPromises: Array<Promise<DH.IIndicatorDataColored>> = indicatorArgs
            .map(async (query, index) => {
                const conceptId = getTableNameFromSql(query);
                if (isError(conceptId)) throw conceptId;
                const concept = await getConceptAsync(`spotlight-${opts.country}`, conceptId);
                const resource: IRAW = resourcesRaw[index][0];
                return {...concept, value: Number(resource.value), year: concept.startYear};
            });
        const resources: DH.IIndicatorDataColored[] = await Promise.all(resourceWithConceptPromises);
        return {
            regionalResources: formatNumbers(resourcesSum, 1),
            regionalResourcesBreakdown: resources
        };
    }
    private async getIndicatorsGeneric(opts: ISpotlightArgs, sqlList: string[])
        : Promise<string[]>  {
        const indicatorArgs: IGetIndicatorArgs[] =
            sqlList.map(query => ({...this.defaultArgs, query, ...opts}));
        const indicatorRaw: IRAW[][] = await Promise.all(indicatorArgs.map(args => getIndicatorData<IRAW>(args)));
       //  if (!format) return indicatorRaw.map(data => indicatorDataProcessingSimple<IProcessedSimple>(data));
        return indicatorRaw.map(data => formatNumbers(data[0].value, 1));
    }
    private async getPopulationDistribution(opts): Promise<DH.IPopulationDistribution[]> {
        const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultArgs,
            query: sql.populationDistribution,
            ...opts
        };
        const data: IRAWPopulationGroup[] = await getIndicatorData<IRAWPopulationGroup>(indicatorArgs);
        return data.reduce((acc: DH.IPopulationDistribution[], row) => {
            const rural = {group: 'rural', value: Number(row.value_rural), year: Number(row.year) };
            const urban = {group: 'urban', value: Number(row.value_urban),  year: Number(row.year) };
            return [...acc, rural, urban];
        }, []);
    }
}
