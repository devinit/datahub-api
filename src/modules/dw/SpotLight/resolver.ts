import { IContext } from '../../../modules';

export default {
    Query: {
        async educationTabRegional(_root, args, ctx: IContext) {
            return ctx.modules.spotLight[args.country].getEducationTabRegional(args);
        },
        async healthTabRegional(_root, args, ctx: IContext) {
            return ctx.modules.spotLight[args.country].getHealthTabRegional(args);
        },
        async localGovernmentFinance(_root, args, ctx: IContext) {
            return ctx.modules.spotLight[args.country].getLocalGovernmentFinance(args);
        },
        async overviewTabRegional(_root, args, ctx: IContext) {
            return ctx.modules.spotLight.getOverViewTabRegional(args);
        },
        async populationTabRegional(_root, args, ctx: IContext) {
            return ctx.modules.spotLight[args.country].getPopulationTabRegional(args);
        },
        async povertyTabRegional(_root, args, ctx: IContext) {
            return ctx.modules.spotLight[args.country].getPovertyTabRegional(args);
        }
    },
    PovertyTabRegional: {
        __resolveType(obj) {
            if (obj.meanExpenditure) {
                return 'PovertyTabKe';
            }
            return 'PovertyTabUg';
        },
    },
    PopulationTabRegional: {
        __resolveType(obj) {
            if (obj.averageDependencyRatio) {
                return 'PopulationTabRegionalUg';
            }
            return 'PopulationTabRegionalKe';
        },
    },
    EducationTabRegional: {
        __resolveType(obj) {
            if (obj.studentsPassRate) {
                return 'EducationTabRegionalUg';
            }
            return 'EducationTabRegionalKe';
        },
    },
    HealthTabRegional: {
        __resolveType(obj) {
            if (obj.districtPerformance) {
                return 'HealthTabRegionalUg';
            }
            return 'HealthTabRegionalKe';
        },
    }
};
