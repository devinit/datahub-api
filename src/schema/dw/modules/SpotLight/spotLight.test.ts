import * as prettyFormat from 'pretty-format';
import SpotLight from '.';
import db from '../../db';

describe.skip('spotlight on uganda tests', () => {
    const spotLight = new SpotLight(db);

    it('should return overview tab data for wakiso district', async () => {
        const data = await spotLight.getOverViewTabRegional({id: 'wakiso', country: 'uganda'});
        expect(prettyFormat(data)).toMatchSnapshot();
    }, 10000);
    it('should return PovertyTab tab data for wakiso district', async () => {
        const data = await spotLight.getPovertyTabRegional({id: 'wakiso', country: 'uganda'});
        expect(prettyFormat(data)).toMatchSnapshot();
    }, 10000);
    it('should return population tab data for wakiso district', async () => {
        const data = await spotLight.getPopulationTabRegional({id: 'wakiso', country: 'uganda'});
        expect(prettyFormat(data)).toMatchSnapshot();
    }, 10000);
    it('should return education tab data for wakiso district', async () => {
        const data = await spotLight.getEducationTabRegional({id: 'wakiso', country: 'uganda'});
        expect(prettyFormat(data)).toMatchSnapshot();
    }, 10000);
    it('should return health tab data for wakiso district', async () => {
        const data = await spotLight.getHealthTabRegional({id: 'wakiso', country: 'uganda'});
        expect(prettyFormat(data)).toMatchSnapshot();
    }, 10000);
    it('should return finance data for wakiso district', async () => {
        const data = await spotLight.getLocalGovernmentFinance({id: 'wakiso', country: 'uganda'});
        expect(prettyFormat(data)).toMatchSnapshot();
    }, 10000);

    afterAll(() => {
       db.$config.pgp.end();
    });
});
