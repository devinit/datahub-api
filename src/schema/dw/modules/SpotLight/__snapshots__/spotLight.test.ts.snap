// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spotlight on uganda tests should return health tab data for wakiso district 1`] = `
"Object {
  \\"districtHealthRank\\": Object {
    \\"toolTip\\": Object {
      \\"heading\\": \\"A composite index based on the performance of all district health indicators. The higher the score the better the rated performance, with 0 the lowest score possible and 100 the highest.\\",
      \\"source\\": \\"Annual health sector performance reports 2010-2014, Ministry of Health.\\",
    },
    \\"value\\": \\"62nd\\",
  },
  \\"districtPerformance\\": Object {
    \\"toolTip\\": Object {
      \\"heading\\": \\"A composite index based on the performance of all district health indicators. The higher the score the better the rated performance, with 0 the lowest score possible and 100 the highest.\\",
      \\"source\\": \\"Annual health sector performance reports 2010-2014, Ministry of Health.\\",
    },
    \\"value\\": \\"70.3\\",
    \\"value_ncu\\": \\"No data\\",
  },
  \\"healthCareFunding\\": Object {
    \\"toolTip\\": Object {
      \\"heading\\": \\"Local government spending to healthcare per person, by district.\\",
      \\"source\\": \\"Local government budgets, Ministry of Finance, Planning and Economic Development.\\",
    },
    \\"value\\": \\"1.5\\",
    \\"value_ncu\\": \\"5.2k\\",
  },
  \\"treatmeantOfTb\\": Object {
    \\"toolTip\\": Object {
      \\"heading\\": \\"Percentage of successes among a number of attempts for treatment of tuberculosis, by district.\\",
      \\"source\\": \\"Annual health sector performance reports 2010-2014, Ministry of Health.\\",
    },
    \\"value\\": \\"70.0\\",
    \\"value_ncu\\": \\"No data\\",
  },
}"
`;

exports[`spotlight on uganda tests should return overview tab data for wakiso district 1`] = `
"Object {
  \\"localGovernmentSpendPerPerson\\": Object {
    \\"toolTip\\": Object {
      \\"heading\\": \\"Total per capita local government expenditure allocations, by district.\\",
      \\"source\\": \\"Uganda budget information, Ministry of Finance, Planning and Economic Development.\\",
    },
    \\"value\\": \\"No data\\",
    \\"value_ncu\\": \\"No data\\",
  },
  \\"poorestPeople\\": Object {
    \\"toolTip\\": Object {
      \\"heading\\": \\"Percentage of people in poverty, in relation to the total population in a given area, based on the national poverty line. It does not take into account the depth of poverty. The data year is given as the document publication date, but is based on estimates from previous household surveys by the Uganda Bureau of Statistics.\\",
      \\"source\\": \\"Profiles of higher (district and municipal) local governments 2014, Uganda Bureau of Statistics.\\",
    },
    \\"value\\": \\"13.1\\",
    \\"value_ncu\\": \\"No data\\",
  },
  \\"regionalResources\\": Object {
    \\"toolTip\\": Object {
      \\"heading\\": \\"The proportion of local government revenue that is locally generated, through collection of fees, fines and other local taxation, by district.\\",
      \\"source\\": \\"Uganda budget information, Ministry of Finance, Planning and Economic Development.\\",
    },
    \\"value\\": \\"4.3m\\",
    \\"value_ncu\\": \\"4.3m\\",
  },
  \\"regionalResourcesBreakdown\\": Array [
    Object {
      \\"data\\": Object {
        \\"color\\": \\"#e8443a\\",
        \\"id\\": \\"spotlight_on_uganda.uganda_igf_resources\\",
        \\"name\\": \\"Locally raised revenues\\",
        \\"uid\\": \\"B1yN1YIsW\\",
        \\"value\\": 863351.789868687,
        \\"year\\": 2015,
      },
      \\"toolTip\\": Object {
        \\"heading\\": \\"The proportion of local government revenue that is locally generated, through collection of fees, fines and other local taxation, by district.\\",
        \\"source\\": \\"Uganda budget information, Ministry of Finance, Planning and Economic Development.\\",
      },
    },
    Object {
      \\"data\\": Object {
        \\"color\\": \\"#eb642b\\",
        \\"id\\": \\"spotlight_on_uganda.uganda_central_resources\\",
        \\"name\\": \\"Centrally raised revenues\\",
        \\"uid\\": \\"Syxy4JYUsZ\\",
        \\"value\\": 3402591.22210666,
        \\"year\\": 2015,
      },
      \\"toolTip\\": Object {
        \\"heading\\": \\"The proportion of local government revenue that is funded though central government transfers, by district.\\",
        \\"source\\": \\"Local government budgets, Ministry of Finance, Planning and Economic Development\\",
      },
    },
    Object {
      \\"data\\": Object {
        \\"color\\": \\"#0089cc\\",
        \\"id\\": \\"spotlight_on_uganda.uganda_donor_resources\\",
        \\"name\\": \\"Donor revenues\\",
        \\"uid\\": \\"BJb1EyFLsW\\",
        \\"value\\": 0,
        \\"year\\": 2015,
      },
      \\"toolTip\\": Object {
        \\"heading\\": \\"Donor revenues.\\",
        \\"source\\": \\"Local government budgets, Ministry of Finance, Planning and Economic Development.\\",
      },
    },
  ],
}"
`;
